### PR TITLE
Send path to svgo

### LINKED ISSUE #60: `svgo.optimize` missing path option
This plugin should send the file path to `svgo`.
The path is an optional, but very useful param for `svgo`. https://github.com/svg/svgo#optimize
Capture d’écran, le 2022-07-07 à 10 52 27
For exemple, the `prefixIds` plugin use the file path to generate unique `id` and class.
In Vue 2, I used `vue-svg-inline-loader` which was correctly setup to do this.
See https://github.com/oliverfindl/vue-svg-inline-loader/blob/28051895a5db3c3beba6025963106566d4cee84a/src/index.js#L234
Capture d’écran, le 2022-07-07 à 10 52 04

### PR DESCRIPTION
**Pull request by mrleblanc101**
Fix #60

### CODE REVIEW

**Comment by mrleblanc101** [comment]
Tested with npm link and work great 👍

**Comment by jpkleemans** [comment]
Hi Sébastien, thanks for your PR! I'll look into it next week.

**Comment by mrleblanc101** [comment]
Thanks

File: index.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 const fs = require('fs').promises
 const { compileTemplate } = require('@vue/compiler-sfc')
-const { optimize: optimizeSvg } = require('svgo')
+const { optimize } = require("svgo");
```
Comment (jpkleemans) [inline index.js:3]: Could you run the linter: `npm run lint`

File: index.js [hunk 2]
```diff
@@ -37,7 +37,10 @@ module.exports = function svgLoader (options = {}) {
       }
 
       if (svgo !== false) {
-        svg = optimizeSvg(svg, svgoConfig).data
+        svg = optimize(svg, {
+          ...svgoConfig,
+          path,
```
Comment (jpkleemans) [inline index.js:42]: Maybe better to turn these around? To make it possible to set the path in the `svgoConfig`

**Review by jpkleemans — CHANGES_REQUESTED** [review]
Thanks for your work. Could you review my comments?
Also, is it possible for you to add a cypress test for this change?

**Comment by mrleblanc101** [comment]
Actually I think path should be passed to `new svgo()` and not `svgo.optimize()` but I wasn't sure if you'd want me to refactor so much.

File: index.js (hunk 1 above)
Comment (mrleblanc101) [reply index.js:3]: Sure, I'll do it this weekend. Didn't notice because there is no pre-commit hook, my bad.

**Comment by mrleblanc101** [comment]
Do you want me to make that change too ? See https://github.com/oliverfindl/vue-svg-inline-loader/blob/28051895a5db3c3beba602596310656
6d4cee84a/src/index.js#L234

**Comment by mrleblanc101** [comment]
> Thanks for your work. Could you review my comments? Also, is it possible for you to add a cypress test for this change?

I added one test

**Comment by jpkleemans** [comment]
> Actually I think path should be passed to `svgo.optimize({ path: 'path.svg' })` alone and the options should be passed to `new svgo(options)` but you don't seem to instantiate the plugin like that.

In the svgo readme, it looks like you can add config to `optimize` just fine: https://github.com/svg/svgo#optimize

**Comment by jpkleemans** [comment]
Thanks for adding the test!
Can you fix the merge conflicts? Then I can merge the PR

**Comment by mrleblanc101** [comment]
@jpkleemans I fixed the merge conflicts.

**Comment by jpkleemans** [comment]
Thank you!

**Pull request merged**